meta-client/src/main/java/org/apache/helix/metaclient/api/ConnectStateChangeListener.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.helix.metaclient.api;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+public interface ConnectStateChangeListener {
+  /**
+   * Called when the connection state has changed. I
+   * @param prevState previous state before state change event.
+   * @param currentState client state after state change event. If it is a one time listsner, it is
+   *                     possible that the metaclient state changes again
+   */
+  void handleConnectStateChanged(MetaClientInterface.ConnectState prevState, MetaClientInterface.ConnectState currentState) throws Exception;
```

Comment:
Curious, why ConnectState is not defined here like other listener implementations?

Reply:
Also we don't have any "handleNewSession" handler so are we going to use same ("handleConnectStateChanged") handler when new connection/session is established? if that's the case then what would be prevState in that case because i don't see any "NOT_CONNECTED" state in ConnectState enum